tests/Http/Controllers/CelebTest.php

```diff
@@ -13,164 +14,334 @@ class CelebTest extends TestCase
+    public function admin_can_update_a_celeb()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->putJson("celebs/$celeb->id", [
+                'name' => 'Jennifer Jaxon',
+                'date_of_birth' => '27/11/1982',
+                'photo' => 'https://generic/photo1 700x600'
+            ])->assertSessionHasNoErrors();
+
+        $this->assertDatabaseHas('celebs', [
+            'name' => 'Jennifer Jaxon',
+            'date_of_birth' => '27/11/1982',
+            'photo' => 'https://generic/photo1 700x600'
         ]);
+    }
+
+    /** @test */
+    public function validation_produces_errors_when_admin_updates_celeb_with_one_data_being_empty()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
-            ->putJson("celebs/$celeb->id")
-            ->assertOk();
+            ->put("celebs/$celeb->id", [
+                'name' => 'Olivia Plankton',
+                'date_of_birth' => '08/04/1994',
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'photo' => 'The photo field is required.'
+            ]);
     }
 
     /** @test */
-    public function normal_user_cannot_access_celebs_update()
+    public function validation_produces_errors_when_admin_updates_celeb_with_all_data_being_empty()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $user = \App\Models\User::factory()->create([
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => null,
+                'date_of_birth' => null,
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'name' => 'The name field is required.',
+                'date_of_birth' => 'The date of birth field is required.',
+                'photo' => 'The photo field is required.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_date_of_birth_after_todays_date()
+    {
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->put("celebs/$celeb->id", [
+                'name' => 'James Islington',
+                'date_of_birth' => '08/03/2999',
+                'photo' => 'https://genericPhotos.com'
+            ])->assertSessionHasErrors([
+                'date_of_birth' => 'The date of birth must be a date before today.'
+            ]);
+    }
+
+    /** @test */
+    public function validation_error_when_admin_updates_celeb_with_name_being_too_short()
```

Comment:
I noticed that the validation is being run for both create and update right? This is okay, but in our case where they both share the same CelebRequest you can just use one endpoint, so you can rename it to something like `validation_errors_when_celeb_name_length_is_invalid` and just choose one of the endpoints. Does that make sense?